sparse/_coo/core.py

```diff
@@ -1451,6 +1451,57 @@ def asformat(self, format, compressed_axes=None):
 
         raise NotImplementedError("The given format is not supported.")
 
+    def pad(self, pad_width, mode="constant", **kwargs):
+        """
+        Performs the equivalent of :obj:`numpy.pad` for :obj:`COO`. Note that
+        this function returns a new array instead of a view.
+
+        Parameters
+        ----------
+        pad_width : {Sequence, Array_like, Int}
+            Number of values padded to the edges of each axis. ((before_1, after_1), … (before_N, after_N)) unique pad widths for each axis. ((before, after),) yields same before and after pad for each axis. (pad,) or int is a shortcut for before = after = pad width for all axes.
+
+        mode : str
+            Pads to a constant value which is fill value. Currently only constant mode is implemented
+
+        Returns
+        -------
+        COO
+            The padded sparse array.
+
+        Raises
+        ------
+        NotImplementedError
+            If mode != 'constant' or there are unknown arguments.
+
+        ValueError
+            If constant_values != self.fill_value
+
+        See Also
+        --------
+        :obj:`numpy.pad` : NumPy equivalent function
+
+        """
+        if mode.lower() != "constant":
+            raise NotImplementedError("Mode '{}' is not supported.".format(mode))
+
+        if kwargs.pop("constant_values", self.fill_value) != self.fill_value:
```

Comment:
This will run into some issues with `nan`s and so forth, use `equivalent` to compare these. Also the default `constant_values` is `0`, not `self.fill_value`. Use `_zero_of_dtype(self.dtype)` to get the default constant value.

Reply:
But Sir in our case shouldn't constant value default be fill value?

Reply:
We want to match the NumPy API as exactly as possible. 😄